### PR TITLE
Refinements to secrets_utils (C4-901)

### PR DESCRIPTION
**Pull request by netsettler**
* Allow identity_name in various secrets_utils operations
* Change a PRINT operation to a logged warning. Improve the wording of the warning.
* Improve some error messages.

### CODE REVIEW

File: dcicutils/secrets_utils.py [hunk 1]
```diff
@@ -56,26 +88,22 @@ def get_identity_secrets(identity_kind: str = GLOBAL_APPLICATION_CONFIGURATION)
...
-            'DecryptionFailureException'  # SM can't decrypt the protected secret text using the provided KMS key.
-            'InternalServiceErrorException',  # An error occurred on the server side.
-            'InvalidParameterException',  # You provided an invalid value for a parameter.
-            'InvalidRequestException',  # Invalid parameter value for the current state of the resource.
-            'ResourceNotFoundException',
-        ]:
-            PRINT('Encountered a known exception trying to acquire secrets.')
+        code = e.response['Error']['Code']
+        if code in _KNOWN_SECRETS_ERRORS:
+            logger.warning(f'Encountered a known exception ({code}) trying to acquire secrets.'
```
Comment (netsettler) [inline dcicutils/secrets_utils.py:93]: I don't know why a `PRINT` was happening, but I figured it should be a log instead. I also am not sure why it needs to happen at all or why it doesn't show the actual error message, but I figured if it was going to hide the error message, then these explanatory comments would be good to change into visible messages. I'm slightly inclined to just show the actual error message... Thoughts from reviewers appreciated.

**Review by willronchetti — APPROVED** [review]
One small comment I think should be addressed

File: dcicutils/secrets_utils.py [hunk 2]
```diff
@@ -26,24 +26,59 @@
...
+    """
+    This evaluates the given environment variable, handling defaults only in the legacy account.
+    """
+    identity_name = os.environ.get(identity_kind)
+    if identity_name:
+        return identity_name
+    account_number = os.environ.get('ACCOUNT_NUMBER')
+    if not account_number or account_number == LEGACY_ACCOUNT_NUMBER:
+        # We do this independent of identity_kind. Is that right? -kmp 31-Aug-2022
+        return LEGACY_DEFAULT_IDENTITY_NAME
```
[the remark this answers is not in the document]
Comment (netsettler) [reply dcicutils/secrets_utils.py:42]: OK, that's just fine. Thanks for simplifying the definition.

[the remark this answers is not in the document]
Comment (netsettler) [reply dcicutils/secrets_utils.py]: Requested changes made and uploaded in commit [c27493a](https://github.com/4dn-dcic/utils/pull/216/commits/c27493a3529d8d1d7d8df8fa42a08193d3765ad9).

**Pull request merged**